upstream.go

```diff
@@ -1298,7 +1298,8 @@ func (uc *upstreamConn) handleMessage(ctx context.Context, msg *irc.Message) err
 		}
 		ch, err := uc.getChannel(name)
```

Comment:
Instead of `getChannel` and having to handle an error, we can do something like

```go
ch := uc.channels.Get(name)
if ch == nil {
    …
}
```

I'd slightly prefer that form because it's clearer when the code-path is taken. `err` could in theory be any runtime error.